src/Cms/HasLock.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Kirby\Data\Data;
+use Kirby\Toolkit\A;
+use Kirby\Toolkit\F;
+
+trait HasLock
+{
+
+    public function lock($user = false, string $language = null)
+    {
+        // read
+        if ($user === false) {
+            $data = $this->content($language)->get('lock');
+
+            if ($data->isEmpty()) {
+                return ['lock' => false];
+            }
+
+            $data = $data->yaml();
+
+            return array_merge($data, [
+                'lock'  => true,
+                'email' => $this->kirby()->user($data['user'])->email(),
+            ]);
+        }
+
+        // write: unset
+        if ($user === null) {
+            return $this->update([
+                'lock' => null
+            ], $language);
+        }
+
+        // write: set
+        return $this->update([
+            'lock' => [
+                'user' => $user,
+                'time' => time()
+            ]
+            ], $language);
+    }
+
+    public function unlock(string $user = null, bool $unset = false, string $language = null)
+    {
+        $data = $this->content($language)->get('unlock')->yaml();
+
+        // read
+        if ($user === null) {
+            return $data;
+        }
+
+        // write: unset
+        if ($unset === true) {
+            $diff = array_diff($data, [$user]);
+
+            if (count($diff) === 0) {
+                $diff = null;
+            }
+
+            return $this->update([
+                'unlock' => $diff
+            ], $language);
+        }
+
+        // write: set
+        return $this->update([
```

Comment:
I think this should delete the `lock` field at the same time. The client would then only need one request and not one to remove the lock and then one to call `unlock` afterwards.